benbarnett02.CodeTrack/TrackDB/TrackingData.cs

```diff
@@ -0,0 +1,211 @@
+    {
+        string format = "dd/MM/yyyy HH:mm tt";
+        using (var myDatabase = new SqliteConnection(connectionString))
+        {
+            myDatabase.Open();
+            var tableCmd = myDatabase.CreateCommand();
+            tableCmd.CommandText =
+                $"UPDATE coding_entries SET Label = '{codeEntry.Label}', StartDate = '{codeEntry.StartDate.ToString(format)}', EndDate = '{codeEntry.EndDate.ToString(format)}' WHERE Id = {codeEntry.Id.ToString()}";
+            tableCmd.ExecuteNonQuery();
+            myDatabase.Close();
+        }
+    }
+    
+
+    // Lol, don't laugh at these next 3 methods I know it's weird and wrong but I had fun.
+    public static void DeleteCodeEntry(CodeEntry codeEntry)
+    {
+        DeleteCodeEntry(codeEntry.Id); // Trigger the Int overload
+    }
+    public static void DeleteCodeEntry(int id)
+    {
+        DeleteCodeEntry(id.ToString()); // Trigger the String overload
+    }
+    
+    public static void DeleteCodeEntry(string id) // This one does the work
+    {
+        using (var myDatabase = new SqliteConnection(connectionString))
+        {
+            myDatabase.Open();
+            var tableCmd = myDatabase.CreateCommand();
+            tableCmd.CommandText =
+            $"DELETE FROM coding_entries WHERE Id = '{id}'";
+            tableCmd.ExecuteNonQuery();
+            myDatabase.Close();
+        }
+    }
+
+
+    public static bool TryGetCodeEntry(string id)
+    {
+        bool exists;
+        using (var myDatabase = new SqliteConnection(connectionString))
+        {
+            myDatabase.Open();
+            var checkCmd = myDatabase.CreateCommand();
+            checkCmd.CommandText = $"SELECT EXISTS(SELECT 1 FROM coding_entries WHERE Id = '{id}')";
+            int checkQuery = Convert.ToInt32(checkCmd.ExecuteScalar());
+
+            if (checkQuery == 0)
+            {
+                exists = false;
+            }
+            else if (checkQuery == 1)
+            {
+                exists = true;
+            }
+            else
+            {
+                throw new Exception($"More than one or less than zero records were found with record ID {id}. Review the database and check for duplicate entries.");
+            }
+            myDatabase.Close();
+        }
+        return exists;
+    }
+
+    // I wanted this to be able to handle ints, as well. And learn about overloading methods.
+    public static bool TryGetCodeEntry(int id)
+    {
+        bool exists = TryGetCodeEntry(id.ToString());
+        return exists;
+    }
+
+    public static CodeEntry GetCodeEntry(string id)
+    {
+        using (var myDatabase = new SqliteConnection(connectionString))
+        {
+            myDatabase.Open();
+            var tableCmd = myDatabase.CreateCommand();
+
+            tableCmd.CommandText =
+                $"SELECT * FROM coding_entries WHERE Id = '{id}'"; ;
+
+            SqliteDataReader reader = tableCmd.ExecuteReader();
```

Comment:
While this isn't as important in SQLite, in SQL Server you'll need an using statement for this reader as well, so you can dispose if this resource. To find out when you need to dispose of a resource, check the implementation (F12 in VS2022), and see if it implements IDisposable. In this case: 
![image](https://user-images.githubusercontent.com/108711476/207208699-34631e14-34b2-43cb-9a65-18793570e641.png)